### DISCUSSION TITLE
Suggestion for snap_manageState Method in MetaMask

### DISCUSSION THREAD

**Discussion by AMIRKHANEF** [MetaMask/snaps#2217] [Ideas]
Hello MetaMask Developers,

I'm currently working on developing a snap, and I came across the `snap_manageState` method. After reviewing the code, I have a suggestion regarding the `ManageStateResult `type.

In the current implementation:
> /**
>  * The result returned by the `snap_manageState` method.
>  *
>  * If the operation is `get`, the result is the state. Otherwise, the result is
>  * `null`.
>  */
> export declare type ManageStateResult = Record<string, Json> | null;

For the **update** operation, the result is **null**. I propose changing this to a **boolean** type, where true indicates a successful operation, and false indicates a failure.
> /**
>  * The result returned by the `snap_manageState` method.
>  *
>  * If the operation is `get`, the result is the state. Otherwise, the result is
>  * `boolean`.
>  */
> export declare type ManageStateResult = Record<string, Json> | boolean;

What are your thoughts on such a change? Do you see potential benefits or drawbacks?

I appreciate your hard work on MetaMask and the support you provide to the developer community. Thank you for considering this suggestion!

**Comment by Montoya** [comment]
Thank you for the suggestion! Right now. `snap_manageState` throws an error if it fails. The best approach is to wrap calls to manageState in `try...catch` blocks to handle errors as needed.

**Reply by AMIRKHANEF** [reply]
Hi, Mr. Montoya,

Thanks for the response! While I understand `snap_manageState` throws errors on failure, I suggest returning **true** for successful updates instead of **null**. I believe, It could add clarity to the outcome and seems more intuitive to me.

BTW, Just a thought I had while digging into the code.

Cheers,

Amir
